src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -409,13 +440,99 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
+                        !span.HasValue ||
+                        _priorityProvider.Priority != CodeActionRequestPriority.Normal)
+                    {
+                        return false;
+                    }
+
+                    Debug.Assert(span.Value.Length < _text.Length);
+
+                    // Condition 3.
+                    // Check if this is a candidate analyzer that can be de-prioritized into a lower priority bucket based on registered actions.
+                    if (!await IsCandidateForDeprioritizationBasedOnRegisteredActionsAsync(analyzer).ConfigureAwait(false))
+                    {
+                        return false;
+                    }
+
+                    // Condition 4.
+                    // We do not want to de-prioritize this analyzer if it reported a diagnostic on a prior document snapshot,
+                    // such that diagnostic's start/end lines intersect the current analysis span's start/end lines.
+                    // If an analyzer reported such a diagnostic, it is highly likely that the user intends to invoke the code fix
+                    // for this diagnostic. Additionally, it is also highly likely that this analyzer will report a diagnostic
+                    // on the current snapshot. So, we deem this as an important analyzer that should not be de-prioritized here.
+                    // Note that we only perform this analysis if the prior document, whose existingData is cached, had same number
+                    // of source lines as the current document snapshot. Otherwise, the start/end lines comparison across
+                    // snapshots is not meaningful.
+                    if (existingData.LineCount == _text.Lines.Count &&
+                        !existingData.Items.IsEmpty)
+                    {
+                        _text.GetLinesAndOffsets(span.Value, out var startLineNumber, out var _, out var endLineNumber, out var _);
+
+                        foreach (var diagnostic in existingData.Items)
+                        {
+                            if (diagnostic.DataLocation.MappedFileSpan.StartLinePosition.Line <= endLineNumber &&
+                                diagnostic.DataLocation.MappedFileSpan.EndLinePosition.Line >= startLineNumber)
+                            {
+                                return false;
+                            }
+                        }
+                    }
+
+                    // 'LightbulbSkipExecutingDeprioritizedAnalyzers' option determines if we want to execute this analyzer
+                    // in low priority bucket or skip it completely. If the option is not set, track the de-prioritized
+                    // analyzer to be executed in low priority bucket.
+                    if (!_owner.GlobalOptions.GetOption(DiagnosticOptionsStorage.LightbulbSkipExecutingDeprioritizedAnalyzers))
+                        _priorityProvider.AddDeprioritizedAnalyzerWithLowPriority(analyzer);
```

Comment:
i would want this strongly doc'ed that the intent here is to both mutate this provider, and ensure that providers forked off it see the mutation.